parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnReaderImpl.java

```diff
@@ -546,6 +573,11 @@ private void initDataReader(Encoding dataEncoding, byte[] bytes, int offset, int
     } catch (IOException e) {
       throw new ParquetDecodingException("could not read page in col " + path, e);
     }
+
+    if (dataEncoding.requiresSequentialReads(writerVersion) && previousReader != null) {
+      // previous reader can only be set if reading sequentially
+      ((RequiresPreviousReader) dataColumn).setPreviousReader(previousReader);
```

Comment:
Discussed offline. If we wanna make it general, we should make encoding to provide a SequentialReader rather than type casting the Reader.
If we wanna make it not that generic, we should not let the "requiresSequentialReads" be part of the normal Encoding's API